### PR TITLE
prov/efa: use local read to copy data to gpu memory

### PR DESCRIPTION
**Pull request by wzamazon**
This PR contain 3 patches that implement local read as the fallback solution to copy data to gpu memory.

### CODE REVIEW

**Comment by wzamazon** [comment]
The first two patches are in separate PR: https://github.com/ofiwg/libfabric/pull/6218 and https://github.com/ofiwg/libfabric/pull/6219

**Comment by wzamazon** [comment]
Update the PR to fix the AWS CI failures.

File: prov/efa/src/rxr/rxr_pkt_type_req.c [hunk 1]
```diff
@@ -823,46 +789,178 @@ ssize_t rxr_pkt_proc_matched_read_rtm(struct rxr_ep *ep,
...
+			rxr_cq_handle_rx_error(ep, rx_entry, err);
+			rxr_pkt_entry_release_rx(ep, pkt_entry);
+			return err;
+		}
+	} else {
+		err = ofi_copy_to_hmem_iov(desc ? desc->peer.iface : FI_HMEM_SYSTEM,
+					   desc ? desc->peer.device.reserved : 0,
+					   rx_entry->iov, rx_entry->iov_count, 0, data, data_size);
+		if (err < 0)
+			return err;
```
Comment (rwespetal) [inline prov/efa/src/rxr/rxr_pkt_type_req.c:960]: Not sure I follow why this error block exits but the earlier one does some cleanup?

File: prov/efa/src/rxr/rxr_pkt_type_req.c [hunk 2]
```diff
@@ -1380,6 +1457,7 @@ void rxr_pkt_handle_eager_rtw_recv(struct rxr_ep *ep,
 		return;
 	}
 
+	ofi_copy_to_iov(rx_entry->iov, rx_entry->iov_count, 0, data, data_size);
```
Comment (rwespetal) [inline prov/efa/src/rxr/rxr_pkt_type_req.c:1460]: Shouldn't this still update `rx_entry->bytes_done`? Same thing for the long rtw message.

**Pull request closed without merging**